src/main/webapp/resources/js/pages/password/PasswordReset.jsx

```diff
@@ -0,0 +1,242 @@
+import React from "react";
+import { render } from "react-dom";
+import { Alert, Button, Col, Form, Input, List, Row, Typography } from "antd";
+import {
+  EyeInvisibleOutlined,
+  EyeTwoTone,
+  LoadingOutlined,
+  LockOutlined,
+} from "@ant-design/icons";
+import { setBaseUrl } from "../../utilities/url-utilities";
+import { blue6 } from "../../styles/colors";
+import { SPACE_MD, SPACE_SM } from "../../styles/spacing";
+import { useSetPasswordMutation } from "../../apis/password-reset";
+import store from "../store";
+import { Provider } from "react-redux";
+
+const { Item } = Form;
+const passwordExpired =
+  new URLSearchParams(window.location.search).get("expired") || false;
+const passwordResetObj = window.PAGE.passwordReset;
+
+/**
+ * React component to render the Password Reset form.
+ * @returns {*}
+ * @constructor
+ */
+function PasswordResetForm() {
+  const [passwordResetForm] = Form.useForm();
+  const [setPassword] = useSetPasswordMutation();
+
+  const [loading, setLoading] = React.useState(false);
+  const [invalidPassword, setInvalidPassword] = React.useState(true);
+  const [updateSuccess, setUpdateSuccess] = React.useState(false);
+  const [updateError, setUpdateError] = React.useState(false);
+  const [errorMessages, setErrorMessages] = React.useState(null);
+  const passwordRef = React.useRef();
+
+  /**
+   * When the component gets added to the page,
+   * focus on the password input input.
+   */
+  React.useEffect(() => {
+    passwordRef.current.focus();
+    passwordRef.current.select();
+  }, []);
+
+  const passwordRules = [
+    i18n("PasswordReset.alert.rule2"),
+    i18n("PasswordReset.alert.rule3"),
+    i18n("PasswordReset.alert.rule4"),
+    i18n("PasswordReset.alert.rule5"),
+    i18n("PasswordReset.alert.rule6"),
+    i18n("PasswordReset.alert.recommendation1"),
+    i18n("PasswordReset.alert.recommendation2"),
+  ];
+
+  const submitPasswordResetForm = () => {
+    setLoading(true);
+    setPassword({
+      resetId: passwordResetObj.id,
+      password: passwordResetForm.getFieldValue("password"),
+    }).then((res) => {
+      if (res.error) {
+        setErrorMessages(res.error.data.errors);
+        setUpdateError(true);
+      } else {
+        setUpdateSuccess(true);
+        setUpdateError(false);
+      }
+      passwordResetForm.resetFields();
+      setLoading(false);
+    });
+  };
+
+  // Validator to check if password meets password policy requirements
+  const validatePassword = (password) => {
+    const passwordRegex =
```

Comment:
Move this to validation-utilities.js

Reply:
Moving this into the validation-utilities file causes the page to not load. In the console it is displaying an error in url-utilities even though it has nothing to do with it. Removing the validation from the file and putting it back in the PasswordReset file makes it work again. Any ideas of what is going on?

Reply:
Weirdest thing:

So in url-utilities we have:

`export function getProjectIdFromUrl(url = window.location.href) {
  const projectIdRegex = /\/projects\/(?<projectId>\d+)/;

  const found = url.match(projectIdRegex);
  return found.groups?.projectId;
}
`

This causes the reset password page to break. In their is an error that the groups is null in the above function.

However changing it to the below the page loads perfectly fine

`export function getProjectIdFromUrl(url = window.location.href) {
  const projectIdRegex = /\/projects\/(?<projectId>\d+)/;

  const found = url.match(projectIdRegex);
  if (found) {
    if (found.groups) {
      return found.groups.projectId;
    }
  }
}`

Reply:
Moved and updated in a1e096c493f11ef384deaa372466b683ca67de7c